core/src/test/java/com/opensymphony/xwork2/conversion/impl/XWorkBasicConverterTest.java

```diff
@@ -228,6 +229,37 @@ public void testPositiveDoubleValue() throws Exception {
         assertEquals(94.1231233d, (Double) convertedObject, 0.0001);
     }
 
+    public void testBigDecimal() {
+        Object convertedObject = basicConverter.convertValue(null, BigDecimal.class);
```

Comment:
@aleksandr-m @lukaszlenart I added a new test that also succeeds for the current master branch.

Conclusion: IMHO we can (should) remove all non-primitive types from the `Map` of `primitiveDefaults` within `DefaultTypeConverter.class b/c they won't ever be used. ;-)

The only access to `primitiveDefaults is in case of `value == null` AND if `toType.isPrimitive()`. This is not the case for `BigDecimal` therefore `null` will be returned -> see L:233-234

Reply:
Even more: non of those types are primitive b/c ALL are class types. Do I miss something? Otherwise the whole map + check could be removed... :confused:

Reply:
Also see https://stackoverflow.com/questions/709961/determining-if-an-object-is-of-primitive-type

Reply:
I would say go ahead and we will see :)

Reply:
@sepe81 :) Nice find. Go ahead.

Reply:
I concluded my new findings within the commit comment of d09d3dc.

Reply:
Well, I'm curious now, why `BigInteger` and `BigDecimal` default to `0` instead of `null`. @lukaszlenart Do you have any idea?

Regarding this PR it looks good. :+1:

Reply:
No idea, this class is one of the oldest and I would like to finish cleaning up the conversion logic at some point :)